core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AzureBatchSource.java

```diff
@@ -32,15 +34,27 @@
 @Name("AzureBlobStore")
 @Description("Batch source to read from Azure Blob Storage.")
 public class AzureBatchSource extends FileBatchSource {
+  private static final Logger LOG = LoggerFactory.getLogger(AzureBatchSource.class);
 
   @SuppressWarnings("unused")
   private final AzureBatchConfig config;
 
   public AzureBatchSource(AzureBatchConfig config) {
-    super(config);
+    super(new FileBatchConfig(config.referenceName, getPrefixedPath(config.account, config.container, config.path),
+                              config.fileRegex, config.timeTable, config.inputFormatClass,
+                              config.fileSystemProperties, config.maxSplitSize, config.ignoreNonExistingFolders,
+                              config.recursive));
     this.config = config;
   }
 
+  // add prefix "wasb://<container>@<account>" to the path if path does not start with "wasb://"
+  private static String getPrefixedPath(String account, String container, String path) {
```

Comment:
Modifying behavior in child classes is cleaner using inheritance rather than modifying properties in a config file. In fact, I think we should refactor how path is handled in the File sources since they all use the description from FileBatchSource, which is inaccurate and confusing.

One way to do this is to create a 'FileSourceConfig' that they all extend that has a getPath() method but no path property. Each source can then have its own path property with its own description, and AzureBatchConfig can do this logic in getPath() instead of creating a modified config object.